opendatahub-io/notebooks
# pre-install minimal requirment python packages on vscode

Related to: https://github.com/opendatahub-io/notebooks/issues/345 

## Description

This PR introduce a minimal list of python packages that are required for data sciences and programming workloads.

**NOTE:** Will follow up a separated issue/pr to incorporate the standard database clients https://github.com/opendatahub-io/notebooks/issues/372

## How Has This Been Tested?
1. Spin up the vscode image generated from this PR
2. Open a terminal inside the container and run `pip list` and ensure there is a long list with packages
3. Test some of the installed packages, by creating a .py or .ipynb file, add the following code:
```
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Create a sample DataFrame
data = {'A': np.random.rand(100),
        'B': np.random.randn(100),
        'C': np.linspace(0, 1, 100)}

df = pd.DataFrame(data)

# Perform some operations using NumPy
df['D'] = np.sin(df['A']) + np.cos(df['B'])

# Plotting with Matplotlib
plt.figure(figsize=(10, 6))

plt.subplot(2, 2, 1)
plt.scatter(df['A'], df['B'])
plt.title('Scatter Plot')

plt.subplot(2, 2, 2)
plt.plot(df['C'], df['D'], color='red')
plt.title('Line Plot')

plt.subplot(2, 2, 3)
plt.hist(df['B'], bins=20, color='green', alpha=0.7)
plt.title('Histogram')

plt.subplot(2, 2, 4)
plt.boxplot(df[['A', 'C']])
plt.title('Boxplot')

plt.tight_layout()
plt.show()
```
4. You shall see the following plot:
![image](https://github.com/opendatahub-io/notebooks/assets/42587738/f33a78dc-76cc-4058-895b-822b0ca93a5d)

5. In case of issue check which interpreter you are using 

## Merge criteria:
